docs/observability/cloud-trace.md

```diff
@@ -0,0 +1,227 @@
+- In-depth Analysis and Visualization: Trace Explorer is the primary tool for analyzing traces, offering visual aids like heatmaps for span duration and line charts for request/error rates. It also provides a spans table, groupable by service and operation, which gives one-click access to representative traces and a waterfall view to easily identify bottlenecks and sources of errors within your agent's execution path
+
+The following example will assume the following agent directory structure
+
+```
+working_dir/
+├── weather_agent/
+│   ├── agent.py
+│   └── __init__.py
+└── deploy_agent_engine.py
+└── deploy_fast_api_app.py
+└── agent_runner.py
+```
+
+```python
+# weather_agent/agent.py
+
+import os
+from google.adk.agents import Agent
+
+os.environ.setdefault("GOOGLE_CLOUD_PROJECT", "{your-project-id}")
+os.environ.setdefault("GOOGLE_CLOUD_LOCATION", "global")
+os.environ.setdefault("GOOGLE_GENAI_USE_VERTEXAI", "True")
+
+
+# Define a tool function
+def get_weather(city: str) -> dict:
+    """Retrieves the current weather report for a specified city.
+
+    Args:
+        city (str): The name of the city for which to retrieve the weather report.
+
+    Returns:
+        dict: status and result or error msg.
+    """
+    if city.lower() == "new york":
+        return {
+            "status": "success",
+            "report": (
+                "The weather in New York is sunny with a temperature of 25 degrees"
+                " Celsius (77 degrees Fahrenheit)."
+            ),
+        }
+    else:
+        return {
+            "status": "error",
+            "error_message": f"Weather information for '{city}' is not available.",
+        }
+
+
+# Create an agent with tools
+root_agent = Agent(
+    name="weather_agent",
+    model="gemini-2.5-flash",
+    description="Agent to answer questions using weather tools.",
+    instruction="You must use the available tools to find an answer.",
+    tools=[get_weather],
+)
+```
+
+## Cloud Trace Setup
+
+### From Agent Engine Deployment
+
+To enable cloud tracing on ADK Agent Engine deployment, you can add `enable_tracing=True` when initialize the `AdkApp` object
+
+```python
+# deploy_agent_engine.py
+
+from vertexai.preview import reasoning_engines
+from vertexai import agent_engines
+from weather_agent.agent import root_agent
+
+import vertexai
+
+PROJECT_ID = "{your-project-id}"
+LOCATION = "{your-preferred-location}"
+STAGING_BUCKET = "{your-staging-bucket}"
+
+vertexai.init(
+    project=PROJECT_ID,
+    location=LOCATION,
+    staging_bucket=STAGING_BUCKET,
+)
+
+adk_app = reasoning_engines.AdkApp(
+    agent=root_agent,
+    enable_tracing=True,
+)
+
+
+remote_app = agent_engines.create(
+    agent_engine=adk_app,
+    extra_packages=[
+        "./weather_agent",
+    ],
+    requirements=[
+        "google-cloud-aiplatform[adk,agent_engines]",
```

Comment:
Could you confirm if users might also need "cloudpickle==3.0"?
See: https://cloud.google.com/vertex-ai/generative-ai/docs/agent-engine/deploy

Reply:
Adk v1.6.1 already include google-cloud-aiplatform v1.103 which already include cloudpicke>=3.0.0. And I already verify it can deploy with no issue